addons/purchase_stock/tests/test_purchase_delete_order.py

```diff
@@ -40,3 +40,53 @@ def test_00_delete_order(self):
         purchase_order_3.button_cancel()
         self.assertEqual(purchase_order_3.state, 'cancel', 'PO is cancelled!')
         purchase_order_3.unlink()
+
+    def test_01_delete_propagation(self):
+        ''' Testcase for deleting purchase order with linked move and propagate cancel off'''
+
+        partner = self.env['res.partner'].create({'name': 'My Partner'})
+        product = self.env['product.product'].create({
+            'name': 'Test',
+            'type': 'product',
+        })
```

Comment:
```suggestion
        self.product_2
```
When possible in tests, always try to make use of already existing records made in `setUpClass()`. It may be little, but added all over the tests it makes quite the difference. :slightly_smiling_face: